<!DOCTYPE html>
<html>
<head>
    <link rel="stylesheet" href="style.css"/>
    <title>Updated Pipeline</title>
    <meta charset="UTF-8"/>
    <meta name="viewpoint" content="width=device-width, initial-scale=1.0"/>
    <!--base href="https..... add in-->
</head>
<!--body begins here -->
<body>
    <div class="grad1">
        <ul class="navi"> <!--navigation bar-->
            <li><a href="homepage.html">Home</a></li>
            <li><a href="rn4_analysis.html">RN4 Analysis</a></li>
            <li><a href="rn6_analysis.html">RN6 Analysis</a></li>
            <li class="dropdown">
                <a class="dropbtn">Pipelines</a>
                <div class="dropdown-content">
                    <a href="original_pipeline.html">Original Pipeline</a>
                    <a class="active" href="updated_pipeline.html">Updated Pipeline</a>
                </div></li>
            <li><a href="software.html">Software</a></li>
            <li><a href="diff_exp.html">Gene Expression</a>
        </ul>
        <h1>Updated Pipeline</h1>
        <div class="clearfix">
            <p>Here displayed is an outline of the updated Tuxedo Protocol (Excluding Differential Expression Analysis by Ballgown as the tools used depend on experiment i.e. no. of samples – see the <a href="http://www.nature.com/articles/nprot.2016.095" title="Pertea <i>et al.</i> 2018." target="_blank">Nature Protocol</a>.)  <br></p>
            <img src="Images/updated_pip_df.png" alt="Updated Pipeline" title="Updated Pipeline" style="float: centre; max-width:100%; height: auto;">
            <p><br>The samples – paired end reads were taken by Hisat2 (grey) and mapped to a pre-built index of the RN4 and RN6 genomes. StringTie (green) was then used for assembly with the aid of a RefSeq annotation. Assembled reads were merged and used as a guide for StringTie -eB to calculate abundance estimation. This results were stored in a .gtf annotation file. This gave values for transcript expression levels in FKPM (Fragments per Kilobase per Million reads) and TPM (Transcripts Per Million). <br><br><b>
            Pertea et al, used Hisat 2 with Stringtie to align, assemble, and estimate abundances for, the sequences from the Rat livers to Rat Genome 4 and 6 from RefSeq. <b/>
            </p>
        </div><br>
    </div>
    <hr>
    <div class="col-lg-2 col-md-2 col-sm-4 col-xs-12">
    <p class="bottom"> This website has been made to display our Analysis of Wang et al, 2014, data with our updated pipeline, as part of an MSc Project at the University of Leicester, UK.<br> <text-align="center">&#9400; 2018</p>
</body>
</html>
